### PR TITLE
Add and integrate Software table

### LINKED ISSUE #528: Add and integrate software table into autoreduction
Issue raised by: [developer]

### What?
As part of the new database design we need to add a table to capture the software used for reduction and the version. As well as adding the database table, we will also need to populate it from the queue processors. This issue should cover both adding the table and ensure it is populated. 

See design image here (where `Job` is current ReductionRun`): 
image

### Where?
Model - `WebApp/autoreduce_webapp/reduction_viewer/models.py`
Integration in QP - `queue_processors/autoreduction_processor/post_process_admin.py` - `reduce.py`

### How?
Part of  move to new database schema

### How to test the issue is resolved
* Will require a unit tests to ensure:
  * That the correct version of Mantid is found
  * That the software (+version) is added (if it does not exist)  Software table
  * That the existing Software object is returned if it already exists
  * That the reduction job is associated with the software it uses

### PR DESCRIPTION
**Pull request by ElliotAOram**
### Summary of work
* Added the Software table to the model
* Added access function for get/create
* Added version find function for mantid
* Propagate version through autoreduction processor to be saved in queue processor

### How to test your work
* Push this code to the dev nodes (will need to perform a database migration) to get the new table
* perform manual submission and ensure that the software records are created and associated with each run

Fixes #528 

**Before merging ensure the release notes have been updated**

### CODE REVIEW

**Comment by ElliotAOram** [comment]
Need to test this manually before it's ready for review

File: model/database/tests/test_access.py [hunk 1]
```diff
@@ -79,6 +79,40 @@ def test_get_experiment(self):
...
 
+    @patch('model.database.access.save_record')
+    def test_get_experiment_create(self, mock_save):
+        """
+        Test: An Experiment record is created
+        When: get_experiment is called with create option True
+        """
+        database = access.start_database()
+        actual = access.get_experiment(rb_number=9999999, create=True)
+        self.assertIsNotNone(actual)
```
Comment (JackEAllen) [inline model/database/tests/test_access.py:90]: Is this test necessary to check if not None? 
Would this not be highlighted in the assert equality checks.

File: WebApp/autoreduce_webapp/reduction_viewer/models.py [hunk 2]
```diff
@@ -46,6 +46,17 @@ def __unicode__(self):
         return u'%s' % self.value
 
 
+class Software(models.Model):
+    """
+    Represents the software used to perform the reduction
+    """
+    name = models.CharField(max_length=100, blank=False)
+    version = models.CharField(max_length=20, blank=False)
```
Comment (JackEAllen) [inline WebApp/autoreduce_webapp/reduction_viewer/models.py:54]: Based on my comment in `model/database/tests/test_access.py` about not None validation, we could alternatively just add not null validation to both columns in the model itself unless there is a need to have just the name or just the version column populated for a given row.

File: queue_processors/autoreduction_processor/post_process_admin.py [hunk 3]
```diff
@@ -359,6 +360,20 @@ def reduce(self):
                         self.message.serialize(limit_reduction_script=True))
             logger.info("Reduction job successfully complete")
 
+    @staticmethod
+    def _get_mantid_version():
```
Comment (JackEAllen) [inline queue_processors/autoreduction_processor/post_process_admin.py:364]: Should we add the return to the docstring for explicite?

**Comment by ElliotAOram** [comment]
@JackEAllen coverall fails sue to lines in untested files so we can ignore this time

**Comment by ElliotAOram** [comment]
@JackEAllen If you restart the queue you should be able to submit again. That bug won't be fixed until the QP is re-written

**Comment by JackEAllen** [comment]
This is now good to merge!  🎉

**Pull request merged**